docs/connecting-data/connecting-to-apis-and-external-data/asana-api.md

```diff
@@ -1,7 +1,177 @@
+
+### **Rate Limits**
+Limits are allocated per authorization token. Limits may vary from token to token.
+
+Free account limit: 150 requests per minute.
+Premium account limit: 1500 requests per minute.
+SearchAPI  are limited to 60 requests per minute.
+Duplication endpoints are limited to 5 concurrent jobs.
+
+## How to Connect DataDisillr to Asana
+To set up a data source connection for Asana, you will need to have:
+
+- A unique name for your data source connection to be used in queries.
+- The API key generated through your Asana account.
+
+### Data Source Form
+
+To locate the Asana form, follow the steps in [Connecting Your Data to DataDistillr](../../). When you get to the window to choose the data source type, select API as shown below.
+
+![Select API from the available choices] [image-3]
+
+On the API screen, select Asana from the list of API types as shown in the image below.
+
+![Select Asana from API types] [image-4]
+
+The following form will appear. Instructions can be found below on how to find the information required to fill each field on the Asana API form.
+
+Once you have filled out all the fields, press the green 'Save' button, and your API will be connected!
+
+![Asana Form] [image-1]
+
+### Name
+Enter any name that will help you recognize this data source from within your query window.
+
+Acceptable characters include:
+
+* lowercase alphanumeric characters
+* underscores
+
+### API Key
+Once you have created a workspace in your Asana account, you can generate the API key that you'll need in order to fill in the form.
+
+Click on your account icon at the top right of your screen. Choose 'My Settings'.
+
+![Click on Asana Account Icon. Choose 'My Settings'.] [image-5]
+
+In your 'My Settings' window, click on the 'Apps' tab.
+
+![Click on the Apps Tab in the 'My Settings' window.] [image-6]
+
+Click on 'Manage Developer Apps' which opens a new tab in your browser.
+
+![Click on 'Manage Developer Apps'.] [image-7]
+
+In the 'Personal access tokens' section, choose 'Create new token'.
+
+![Click on 'Create new token'.] [image-8]
+
+Enter a descriptive name for your token and check the checkbox to agree to the API terms and conditions. Click 'Create token'.
+
+![Enter name and check the checkbox. Click 'Create token'.] [image-9]
+
+Click 'Copy' to copy the token to your clipboard.
+
+![Click 'Copy' to copy the token.] [image-10]
+
+Paste the token into the 'API Key' field in the Asana form.
+
+## Endpoints
+Please see [Asana's API Reference](https://developers.asana.com/docs/asana) for more on Asana's endpoints.
+
+The table below shows a list of endpoints available to connect to within the DataDistillr application. If you need to connect to any endpoints not listed in the table below, please use the [Custom APIs](https://docs.datadistillr.com/connecting-data/connecting-to-apis-and-external-data/custom-apis/) Form.
+
+| Endpoint | Required | Optional |
```

Comment:
For consistency, add a description section to this table

Reply:
Where should I get the descriptions from?